crates/shared/src/event_handling.rs

```diff
@@ -111,6 +112,29 @@ where
         }
     }
 
+    /// Creates a new instance of the event handler that does not index events appearing in blocks before the specified
+    /// input date. Note that this is a different behavior compared to [`Self::new()`]: that function always restarts
+    /// indexing from the specified input block on creation; this function only indexes from the specified input block
+    /// if there are no more recent events in the database.
+    pub async fn new_skip_blocks_before(
+        block_retriever: Arc<dyn BlockRetrieving>,
+        contract: C,
+        store: S,
+        skip_blocks_before: BlockNumberHash,
+    ) -> Result<Self> {
+        let last_handled_block = store.last_event_block().await?;
+        Ok(Self::new(
+            block_retriever,
+            contract,
+            store,
+            if last_handled_block >= skip_blocks_before.0 {
```

Comment:
I believe if you would move this logic into `determine_ethflow_indexing_start()` you wouldn't need any new way to instantiate the `EventUpdater` or `EventHandler` (`new_skip_blocks_before()`).

Reply:
Imho this is something we might need in the future for other contracts and not custom configuration for the eth-flow. On the other hand, if you prefer I can move this logic and it should work.

Reply:
I'm just a bit hesitant to introduce this kind of specialization which could also be handled with the general case.
You could also consider dropping the changes to `EventHandler::new_skip_blocks_before()` and make the block selection logic part of `EventUpdated::new_skip_blocks_before()`. WDYT?

Reply:
>Imho this is something we might need in the future for other contracts and not custom configuration for the eth-flow.

do you have a concrete example in mind? 
Otherwise, I would also slightly lean putting the logic into determine_ethflow_indexing_start()

(but this is only a slight preference. should not block merging)